qiskit_ibm_provider/transpiler/passes/scheduling/scheduler.py

```diff
@@ -0,0 +1,264 @@
+                #
+                #           |t0q
+                # Q ▒▒▒▒▒▒▒▒▒░░
+                # C ▒▒▒░░░░░░░░
+                #     |t0c
+                #
+                # In this case, you can insert readout access before tq0
+                #
+                #           |t0q
+                # Q ▒▒▒▒▒▒▒▒▒▒▒
+                # C ▒▒▒░░░▒▒░░░
+                #         |t0q
+                #
+                t0c = max(t0q, t0c)
+            t1c = t0c
+            for bit in node.op.condition_bits:
+                # Lock clbit until state is read
+                self._idle_after[bit] = (self._current_block_idx, t1c)
+
+            # It starts after register read access
+            t0 = max(t0q, t1c)  # pylint: disable=invalid-name
+
+            t1 = t0 + op_duration  # pylint: disable=invalid-name
+            self._update_idles(node, t0, t1)
+
+            # Terminate the conditional block
+            self._begin_new_circuit_block()
+
+        else:
+            # Fall through to generic case if not conditional
+            self._visit_generic(node)
+
+    def _visit_measure(self, node: DAGNode) -> None:
+        """Visit a measurement node.
+
+        Measurement currently triggers the end of a pulse block in IBM dynamic circuits hardware.
+        This means that it is possible to schedule *up to* a measurement (and during its pulses)
+        but the measurement will be followed by a period of indeterminism.
+        All measurements on disjoint qubits will be collected on the same qubits
+        to be run simultaneously."""
+        current_block_measure_qargs = self._current_block_measure_qargs()
+        measure_qargs = set(node.qargs)
+
+        t0q = max(
+            self._idle_after[q][1] for q in measure_qargs
+        )  # pylint: disable=invalid-name
+
+        # If the measurement qubits overlap, we need to start a new scheduling block.
+        if current_block_measure_qargs & measure_qargs:
+            self._begin_new_circuit_block()
+            t0q = 0
+        # Otherwise we need to increment all measurements to start at the same time within the block.
+        else:
+            t0q = max(  # pylint: disable=invalid-name
+                itertools.chain(
+                    [t0q],
+                    (
+                        self._node_start_time[measure][1]
+                        for measure in self._current_block_measures
+                    ),
+                )
+            )
+
+        # Insert this measure into the block
+        self._current_block_measures.add(node)
+
+        # now update all measure qarg times.
+
+        self._current_block_measures.add(node)
+
+        for measure in self._current_block_measures:
+            t0 = t0q  # pylint: disable=invalid-name
+            bit_indices = {bit: index for index, bit in enumerate(self._dag.qubits)}
+            measure_duration = self.durations.get(
+                Measure(), [bit_indices[qarg] for qarg in node.qargs], unit="dt"
```

Comment:
Yes, different qubits can have different measurement lengths although its rarely used. Importantly all parallel measurements will start but not necessarily finish together.

Reply:
No, I'm asking if measure operation depends on combination of qubits. i.e.
```python
qc1 = QuantumCircuits(2, 2)
qc1.measure(0, 0)
qc1.barrier()
qc1.measure(1, 0)

qc2 = QuantumCircuits(2, 2)
qc2.measure(0, 0)
qc2.measure(1, 0)
```
could yield different duration? Current code assumes backend will provide measure duration for every combination of qubits. Likely inst duration object suffers from the combinatorial explosion. This is bit against current implementation of `Measure` instruction at logical circuit layer, where it always takes a single qubit. It implicitly assumes operation is independent. In this sense, this code should be
```python
measure_duration = max(self.durations.get(Measure(), bit_indices[qarg]) for qarg in node.qargs)
```
unless backend provide dedicated multi-qubit measurement instruction. Perhaps I'm missing something.

Reply:
The duration of the amalgamated measurement will depend on the qubits being measured. If qubits x & y are measured the duration of the amalgamated "measurement" operation will be `max(x,y)`, however, each qubit will only be captured for the specified duration (and can have a similarily varying acquisition pulse).

Reply:
Ah likely I misunderstood the behavior of the code. Since this iterates over all measurements in the block **individually**, it doesn't change duration depending on the combination of qubits (and `node.qargs` here is a single qubit in our device configuration). This is another question, why you cannot use `_get_duration` method? This also takes pulse gate into account, and we can in principle override measurement with the pulse gate.